### ISSUE TITLE
json_query does not send token in HTTP request

### ISSUE THREAD

**Issue by noreplyavailable** [influxdb-rs/influxdb-rust#126]
-   Rust Version: stable-x86_64-pc-windows-msvc | rustc 1.71.0 (8ede3aae2 2023-07-12)
-   Runtime version: Tokio 1.29.1
-   OS Version: Windows 10 21H2
-   InfluxDB version: 2.7 - Docker

Running the following code:
```
    pub async fn read_measurement(&self, query_str: &str) -> Result<(), Error> {
        let read_query = ReadQuery::new(query_str);
        let mut result = self.client.json_query(read_query).await?;

        result.deserialize_next::<Measurement>()?
            .series.into_iter()
            .map(|mut value| {
                println!("{:?}", value);
        }).len(); // temporary

        Ok(())
    }
```

Will result with this error:
`thread 'main' panicked at 'called Result::unwrap() on an Err value: AuthorizationError', src\main.rs:51:81`

Which would make sense because looking at the raw packet using Wireshark shows that no token is being included in the HTTP request:
![image](https://github.com/influxdb-rs/influxdb-rust/assets/98525816/af44ba19-900a-4a33-902f-0fc70795b31a)

Causing an `Unauthorized` response:
![image](https://github.com/influxdb-rs/influxdb-rust/assets/98525816/1856557c-6033-4d5d-8569-960458768d40)

---
As apposed to write requests which work fine:
![image](https://github.com/influxdb-rs/influxdb-rust/assets/98525816/ec556de1-bd9c-46d8-9100-cd0fefd22dee)
![image](https://github.com/influxdb-rs/influxdb-rust/assets/98525816/0ad961d9-a31f-4705-a4a4-7bcaeaed7c08)

And regular read requests which also work fine (using `query` instead of `query_json`):
![image](https://github.com/influxdb-rs/influxdb-rust/assets/98525816/44875b4b-5f6e-4a23-a7e8-1abc78af2f22)
![image](https://github.com/influxdb-rs/influxdb-rust/assets/98525816/39993cbd-f12e-4dfb-92e9-dc913993692f)

Code used for `read`:
```
    pub async fn read(&self, query_str: &str) -> Result<String, Error> {
        let read_query = ReadQuery::new(query_str);
        Ok(self.client.query(read_query).await?)
    }
```

**Comment by msrd0** [comment]
Can you share the code that creates your `self.client`? Did you try calling [`with_token`](https://docs.rs/influxdb/latest/influxdb/struct.Client.html#method.with_token) on the client with the token from your successful request?

**Comment by noreplyavailable** [comment]
```
    pub async fn new(host: &str, db_name: &str, token: &str) -> Result<Self, Error> {
        let client = Client::new(
            format!("http://{}", host), 
            db_name
        ).with_token(token);

        client.ping().await?; // make sure connection is there

        Ok(InfluxDbConnection{
            client
        })

    }
```

**Comment by msrd0** [comment]
Possibly related to #123 - CC @Empty2k12

**Comment by skreger-datahaven** [comment]
I'm having the same issue using InfluxDB Cloud Serverless (version 3). I was wondering if it's a major version incompatibility (V3) but apparently the token implementation isn't working correctly.

**Comment by msrd0** [comment]
Should be fixed by #128
